modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4DefaultHttpRequest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.http.netty4;
+
+import org.opensearch.core.common.bytes.BytesArray;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.http.HttpRequest;
+import org.opensearch.rest.RestRequest;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import io.netty.handler.codec.http.DefaultHttpRequest;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.cookie.Cookie;
+import io.netty.handler.codec.http.cookie.ServerCookieDecoder;
+import io.netty.handler.codec.http.cookie.ServerCookieEncoder;
+
+public class Netty4DefaultHttpRequest extends AbstractNetty4HttpRequest implements HttpRequest {
```

Comment:
Its referenced in the companion security PR (https://github.com/opensearch-project/security/pull/3418/files#diff-62eb3270a6f8cc2b9e7b4a92393d1ed9c3ed291ba7548cfddfbe1140e715b0d1R91). I created a separate class to make it clear that it was differentiated from the Netty4HttpRequest which takes a FullHttpRequest in its constructor and has content. This object is like a Netty4HttpRequest, but has no methods around the content of the request. 

It may be possible to add a separate constructor to Netty4HttpRequest to take in DefaultHttpRequest (a netty class that has all request data except for the content), but it could get confusing. I think a separate class where both inherit from the abstract makes sense.

Reply:
We don't need to have 2 requests,  the `Netty4HttpRequest ` should be sufficient. This class is equivalent to Netty4HttpRequest with empty content, right?

Reply:
@reta I changed the implementation `Netty4HttpRequest` to take in a generate `HttpRequest` (netty class)

There are 2 separate HttpRequest classes which causes confusion:

- The netty HttpRequest interface
- org.opensearch.http.HttpRequest

I updated the `Netty4HttpRequest` to be more generic, but I am not in love with the changes.

Let me know what you think.

Reply:
@peternied ^ 

I updated the `Netty4HttpRequest` to be generic. lmk what you think.